Mathlib/Data/Int/Defs.lean

```diff
@@ -264,6 +264,15 @@ protected theorem le_induction_down {P : ℤ → Prop} {m : ℤ} (h0 : P m)
     fun k hle hi _ ↦ h1 k hle (hi hle)
 #align int.le_induction_down Int.le_induction_down
 
+/-- A strong recursor for `Int` that specifies explicit values for integers below a threshold,
+and is analogous to `Nat.strongRec` for integers on or above the threshold. -/
+def strongRec {P : ℤ → Sort*} (m : ℤ) (h0 : ∀ n < m, P n)
+    (h1 : ∀ n, m ≤ n → (∀ k, k < n → P k) → P n) (n : ℤ) : P n :=
+  (em <| n < m).by_cases (h0 _) fun h ↦ h1 _ (Int.not_lt.mp h) (n.inductionOn' m h0
+    (fun _n _ hn l lt1 ↦ (em <| l < m).by_cases (h0 _) fun nlt ↦ h1 _ (Int.not_lt.mp nlt)
+      fun _k lt ↦ hn _ <| Int.lt_of_lt_of_le lt <| lt_add_one_iff.mp lt1)
+    fun _n _ hn _l lt ↦ hn _ <| Int.lt_trans lt <| pred_self_lt _)
```

Comment:
To be consistent with the naming of the other recursors in this file, and minor golf:
```suggestion
@[elab_as_elim] protected def strongInduction {P : ℤ → Sort*} (m : ℤ) (lt : ∀ n < m, P n)
    (ge : ∀ n ≥ m, (∀ k : ℤ, k < n → P k) → P n) (n : ℤ) : P n :=
  (em <| n < m).by_cases (lt n) fun h ↦ ge n (not_lt.mp h) <| n.inductionOn' m lt
    (fun _ _ hn l lt1 ↦ (em <| l < m).by_cases (lt l) fun nlt ↦ ge l (not_lt.mp nlt)
      fun n lt2 ↦ hn n <| lt_of_lt_of_le lt2 <| lt_add_one_iff.mp lt1)
    fun n _ hn l lt1 ↦ hn l <| lt1.trans n.pred_self_lt
```
Note that I am very much against this naming, but we should change them in a future PR.

Personally I think this is too terse - maybe pattern-matching using `induction` would be easier to read.

Reply:
Note that Lean doesn't know ℤ is a linear order in this file, so we cannot use `not_lt`, `lt_of_lt_of_le` and `lt_trans` and have to resort to the corresponding `Int.` lemmas, which are protected.